### ISSUE TITLE
Dual API convolution: déprécier Tensor→Tensor paramétrique au profit de Module→Module

### ISSUE THREAD

**Issue by Ouaziz-chedli** [PoCInnovation/MLBlock#14]
## Contexte

`backend/mlblock/blocks/convolution-6366F1/` expose **2 APIs incompatibles** dans un même dossier (16 blocks) :

**API Tensor (eager) — 11 blocks**
```python
# conv2d.py / linear.py / conv1d/conv3d/conv_transpose2d/embedding/dropout/flatten/upsample
def conv2d(in_1: "torch.Tensor", in_channels: int, out_channels: int, ...) -> "torch.Tensor":
    return nn.Conv2d(in_channels=in_channels, out_channels=out_channels, ...)(in_1)

def linear(in_1: "torch.Tensor", in_features: int, out_features: int, ...) -> "torch.Tensor":
    return nn.Linear(in_features=in_features, out_features=out_features, ...)(in_1)
```
- `in_1: torch.Tensor` required 1er arg → `family=tensor` (`core/types.py:family_of`)
- Codegen `core/generator.py:229` : `out_N = conv2d(in_1=out_M, ...)`

**API Module (composable) — 5 blocks**
```python
# conv2d_layer.py / linear_layer.py / flatten_layer.py / relu_layer.py / maxpool2d_layer.py / rnn_layer.py
def conv2d_layer(in_channels: int, out_channels: int, in_1: "torch.nn.Module" = None, ...) -> "torch.nn.Module":
    layer = nn.Conv2d(in_channels=in_channels, out_channels=out_channels, ...)
    return nn.Sequential(in_1, layer) if in_1 is not None else layer

def rnn_layer(input_size: int, hidden_size: int, in_1: "torch.nn.Module" = None) -> "torch.nn.Module":
    layer = _RNNOut(nn.RNN(input_size=input_size, hidden_size=hidden_size, batch_first=True))
    return nn.Sequential(in_1, layer) if in_1 is not None else layer
```
- `in_1: torch.nn.Module = None` optionnel dernier arg → `family=module`
- Codegen identique mais variable = Module : `m2 = relu_layer(in_1=m1)`

Validation `validation.py:classify` + front `utils/typeCheck.ts` → `tensor` vs `module` = `incompatible` (mix `conv2d → linear_layer` rejeté).

Dossier mal nommé : contient `linear/flatten/rnn` génériques, pas que de la convolution. Collision couleur `#F59E0B` avec `modeles-F59E0B`.

## Problème

1. **Poids éphémères (bug d'entraînement)** : `nn.Conv2d(...)(x)` alloue + init aléatoire à chaque forward, jeté après. L'optimizer (`adam.py:4`, `sgd.py:4` : `torch.optim.Adam(in_1.parameters())`) capture des paramètres morts → `train_model.py:6` (`in_1: torch.nn.Module` + `.to(device)/.train()/loss.backward()`) ne peut pas s'y brancher, `model_checkpoint.py` (`state_dict`) pareil.
2. **Perf** : N allocations `weight [out×in×k×k]` + init Kaiming par forward vs 1× pour Module.
3. **Incohérence catalogue** : `activation-EF4444/relu` n'existe qu'en Tensor alors que `convolution/relu_layer` existe en Module ; utilisateur doit deviner.
4. **Casses** : `nn.Conv1D/Elu/Gelu/Batchnorm1D/Avgpool2D` → crash runtime (jamais validé par `registry.py`).

## Pourquoi préférer `nn.Module` (`_layer`)

- **Contrat PyTorch** : `Module` = état (poids, buffers, `training` flag). `Tensor` = donnée. Une convolution est paramétrique → doit être un Module.
- **Entraînement correct** : poids persistants, gradients accumulés sur `epochs`, `optimizer.step()` cohérent, `Sequential` déplaçable `.to(cuda/mps)`, sauvegardable.
- **Composabilité** : `m = linear_layer(128,10, flatten_layer(relu_layer(conv2d_layer(3,16))))` → un seul Module à passer à `train_model`/`train_epoch`/`evaluate`.
- `Tensor` reste légitime uniquement pour ops **stateless** (`flatten`, `relu` sans poids) où `F.relu(x)` serait même préférable à `nn.ReLU()(x)`, et pour le bloc virtuel `input(shape)`.

## Proposition

1. **Renommer** `convolution-6366F1` → `layers-6366F1` (ou `nn-6366F1`) — maj `registry.py:_color_from_folder`.
2. **Déprécier** les 7 blocks Tensor paramétriques : `conv1d/conv2d/conv3d/conv_transpose2d/linear/embedding` (garder `flatten/dropout/upsample` en Tensor **et** Module si besoin, mais corriger en `F.*`).
3. **Migrer** exemples/docs (`configs/cnn_mnist.json`, `scripts/generate_samples.py`) vers `*_layer`.
4. **Corriger casses** `nn.*` + ajouter test `registry → import + forward smoke` pour éviter régression.
5. Option: wrapper compat temporaire qui log `DeprecationWarning` et délègue au `*_layer` via `_RNNOut`-like adaptateur Tensor→Module.

## Critères d'acceptation

- [ ] Plus de block `Tensor` avec poids dans le catalogue (ou marqué `deprecated` + warning)
- [ ] Un pipeline CNN MNIST complet s'écrit uniquement en `*_layer` + `train_model` + `adam` + `cross_entropy_loss` et passe `validation.validate` + `generator.generate_code` + exécution locale
- [ ] `ruff` + `pytest mlblock/tests -q` + `pnpm build` verts

## Références

- `backend/mlblock/blocks/convolution-6366F1/*.py`
- `backend/mlblock/core/types.py:12-45` (`family_of`), `validation.py:83-104` (`classify`)
- `backend/mlblock/core/generator.py:174-249` (`generate_code`)
- `backend/mlblock/blocks/entrainement-DE497D/train_model.py:6`, `adam.py:4`, `model_checkpoint.py:4`

**Comment by Ouaziz-chedli** [comment]
Resolved via #18, merged into `spec/15-ux-blocks-rework` in commit c7d173d.
- Removed ephemeral tensor operation blocks with weights (`conv1d`, `conv2d`, `conv3d`, `conv_transpose2d`, `linear`).
- Replaced with canonical composable `torch.nn.Module` blocks (`*_layer.py`).
- Renamed category `convolution-6366F1` to `layers-6366F1`.
- Added backward compatibility adapters and catalog lookup in `mlblock/core/adapters.py` and `mlblock/blocks/registry.py`.
- Updated `configs/cnn_mnist.json` and all backend/frontend test suites.
